com.unity.render-pipelines.universal/Runtime/2D/LightUtility.cs

```diff
@@ -170,127 +171,250 @@ public static Bounds GenerateSpriteMesh(Mesh mesh, Sprite sprite)
         }
 
-        public static List<Vector2> GetFalloffShape(Vector3[] shapePath)
+        static bool LineIntersection(float epsilon, float2 p1, float2 p2, float2 p3, float2 p4, ref float2 result)
         {
-            var extrusionDir = new List<Vector2>();
-            for (var i = 0; i < shapePath.Length; ++i)
+            float bx = p2.x - p1.x;
+            float by = p2.y - p1.y;
+            float dx = p4.x - p3.x;
+            float dy = p4.y - p3.y;
+            float bDotDPerp = bx * dy - by * dx;
+            if (math.abs(bDotDPerp) < epsilon)
+            {
+                return false;
+            }
+            float cx = p3.x - p1.x;
+            float cy = p3.y - p1.y;
+            float t = (cx * dy - cy * dx) / bDotDPerp;
+            if ((t >= -epsilon) && (t <= 1.0f + epsilon))
+            {
+                result.x = p1.x + t * bx;
+                result.y = p1.y + t * by;
+                return true;
+            }
+            return false;
+        }        
+        
+        static NativeArray<float4> GetFalloff(Vector3[] shapePath, float fallOff, Color meshInterior, ref List<ContourVertex> vertices)
+        {
+            var shapePathLength = shapePath.Length;
+            var extrusionDir = new NativeArray<float2>(shapePathLength, Allocator.Temp);
+            var smoothInputs = new NativeArray<float4>(shapePathLength * 3, Allocator.Temp);
+            var bi = 0;
+            
+            for (var i = 0; i < shapePathLength; ++i)
             {
-                var h = (i == 0) ? (shapePath.Length - 1) : (i - 1);
                 var j = (i + 1) % shapePath.Length;
-
-                var pp = shapePath[h];
+                
                 var cp = shapePath[i];
                 var np = shapePath[j];
-
-                var cpd = cp - pp;
-                var npd = np - cp;
-                if (cpd.magnitude < 0.001f || npd.magnitude < 0.001f)
+                
+                float3 npd = np - cp;
+                if (math.length(npd) < 0.001f)
                     continue;
+                
+                var vr = math.normalize(npd);
+                var vn = new float2(-vr.y, vr.x);
 
-                var vl = cpd.normalized;
-                var vr = npd.normalized;
-
-                vl = new Vector2(-vl.y, vl.x);
-                vr = new Vector2(-vr.y, vr.x);
-
-                var va = vl.normalized + vr.normalized;
-                var vn = -va.normalized;
-
-                if (va.magnitude > 0 && vn.magnitude > 0)
+                var va = -math.normalize(vn);
+                var dir = new float2(va.x, va.y);
+                extrusionDir[i] = dir;
+            }
+            
+            for (var i = 0; i < shapePathLength; i++)
+            {
+                var j = (i == shapePath.Length - 1) ? 0 : (i + 1);
+                var sPI = shapePath[i];
+                var sPJ = shapePath[j];
+                var eDI = extrusionDir[i];
+                var eDJ = extrusionDir[i];
```

Comment:
Is this intentional? eD**J** = extrusionDir[**i**];

I tried changing it to eDJ = extrusionDir[j], but the result was wrong.

Reply:
eDJ = extrusionDir[i]; is intentional.. Just was testing different values.

Reply:
We have a better implementation now based on Clipper. The above is not needed.